tutorial/source/enumeration.ipynb

```diff
@@ -0,0 +1,516 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can learn the global parameters using SVI with an autoguide."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Running for 10 time steps\n",
+      "x_0.shape = torch.Size([])\n",
+      "x_1.shape = torch.Size([])\n",
+      "x_2.shape = torch.Size([])\n",
+      "x_3.shape = torch.Size([])\n",
+      "x_4.shape = torch.Size([])\n",
+      "x_5.shape = torch.Size([])\n",
+      "x_6.shape = torch.Size([])\n",
+      "x_7.shape = torch.Size([])\n",
+      "x_8.shape = torch.Size([])\n",
+      "x_9.shape = torch.Size([])\n",
+      "Running for 10 time steps\n",
+      "x_0.shape = torch.Size([10, 1])\n",
+      "x_1.shape = torch.Size([10, 1, 1])\n",
+      "x_2.shape = torch.Size([10, 1, 1, 1])\n",
+      "x_3.shape = torch.Size([10, 1, 1, 1, 1])\n",
+      "x_4.shape = torch.Size([10, 1, 1, 1, 1, 1])\n",
+      "x_5.shape = torch.Size([10, 1, 1, 1, 1, 1, 1])\n",
+      "x_6.shape = torch.Size([10, 1, 1, 1, 1, 1, 1, 1])\n",
+      "x_7.shape = torch.Size([10, 1, 1, 1, 1, 1, 1, 1, 1])\n",
+      "x_8.shape = torch.Size([10, 1, 1, 1, 1, 1, 1, 1, 1, 1])\n",
+      "x_9.shape = torch.Size([10, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1])\n"
+     ]
+    }
+   ],
+   "source": [
+    "hmm_guide = AutoDiagonalNormal(poutine.block(hmm_model, expose=[\"transition\", \"emission\"]))\n",
+    "\n",
+    "pyro.clear_param_store()\n",
+    "elbo = TraceEnum_ELBO(max_plate_nesting=1)\n",
+    "elbo.loss(hmm_model, hmm_guide, data, data_dim=data_dim);"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Notice that the model was run twice here: first it was run without enumeration by `AutoDiagonalNormal`, so that the autoguide can record all sample sites; then second it is run by `TraceEnum_ELBO` with enumeration enabled. We see in the first run that samples have the standard interpretation, whereas in the second run samples have the enumeration interpretation.\n",
+    "\n",
+    "For more complex examples, including minibatching and multiple plates, see the [HMM tutorial](https://github.com/uber/pyro/blob/dev/examples/hmm.py)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### How to enumerate more than 25 variables <a class=\"anchor\" id=\"How-to-enumerate-more-than-25-variables\"></a>\n",
+    "\n",
+    "PyTorch tensors have a dimension limit of 25 in CUDA and 64 in CPU. By default Pyro enumerates each sample site in a new dimension. If you need more sample sites, you can annotate your model with  [pyro.markov](http://docs.pyro.ai/en/dev/poutine.html#pyro.poutine.markov) to tell Pyro when it is safe to recycle tensor dimensions. Let's see how that works with the HMM model from above. The only change we need is to annotate the for loop with `pyro.markov`:\n",
```

Comment:
"annotate the for loop with `pyro.markov`" add: because the sample sites in each iteration of the for loop only depend on those in the previous iteration, or something like that